app/presenters/submission_presenter.rb

```diff
@@ -33,4 +34,13 @@ def previous_to_rate
   def submission
     __getobj__
   end
+
+  def average(rates)
+    count = rates.count
+    if count == 0
+      0
+    else
+      (rates.sum(:value).to_f / count)
```

Comment:
I know you just refactored existing code and probably there is no point in changing this now, but for future: such things should be calculated on the db level.